tests/vector/test_spaces.py

```diff
@@ -95,6 +95,32 @@ def test_batch_space(space, expected_batch_space_4):
     assert batch_space_4 == expected_batch_space_4
 
 
+def test_batch_space_seed():
+    for space in [
+        Box(0, 10, seed=123),
+        Tuple([Box(0, 5, seed=123), Box(0, 3, seed=123)], seed=123),
+        Dict(
+            {"space-1": Box(0, 5, seed=123), "space-2": Box(0, 10, seed=123)}, seed=123
+        ),
+    ]:
+        batched_space = batch_space(space)
+        assert space.np_random == batched_space.np_random
+        for _ in range(100):
+            assert np.all(space.sample() == batched_space.sample())
```

Comment:
Wait a minute, the compared spaces won't have the same number of dimensions, so this equality check uses broadcasting, right? Does this still work when comparing dicts? If so, that's something I wasn't aware of!

Reply:
I only did this testing for n=1 as then the sample values should have the same size, though sometimes with an additional dimension (numpy can deal with that fine). 
I haven't done any testing to check if the equivalence check if ok when you have n>1

Reply:
Just to make sure I understand: Is your goal with this PR to make `stack([space.sample() for _ in range(5)]) == `batch_space(space, 5).sample()` ?
If so, that's neat! I suggest you test it like this, and with dicts and weird spaces as well.
The functions in here might be helpful for the nested spaces: https://numpy.org/doc/stable/reference/routines.testing.html

Reply:
Sadly not really, that I don't believe is possible due to the random number generator being updated every time that sample is called meaning that for all batched space if n > 1 then the samples are not equivalent. The PR is sadly quite a bit simpler, it is that the batched space currently ignores the original space's random number seed and this PR fixes that